File: app/src/main/java/com/bth/running/weather/WeatherApi.java
package com.bth.running.weather;

import com.bth.running.weather.model.Weather;

import retrofit2.http.GET;
import retrofit2.http.Query;
import rx.Observable;

/**
 * @author Martin Macheiner
 *         Date: 21.09.2017.
 */

public interface WeatherApi {

    String SERVICE_ENDPOINT = "https://api.openweathermap.org/data/2.5/";

    String API_KEY = "f1a5564dfc20059ace98a12dafc792d8";

    @GET("forecast")
    Observable<Weather> getWeatherForecast(@Query("q") String place,
                                           @Query("appid") String apiKey,
                                           @Query("units") String units);

}

File: app/src/main/java/at/shockbytes/corey/running/weather/WeatherApi.kt
package at.shockbytes.corey.running.weather

import at.shockbytes.corey.running.weather.model.Weather
import io.reactivex.Observable

import retrofit2.http.GET
import retrofit2.http.Query

/**
 * @author Martin Macheiner
 * Date: 21.09.2017.
 */

interface WeatherApi {

    @GET("forecast")
    fun getWeatherForecast(@Query("q") place: String,
                           @Query("appid") apiKey: String,
                           @Query("units") units: String): Observable<Weather>

    companion object {

        const val SERVICE_ENDPOINT = "https://api.openweathermap.org/data/2.5/"

        const val API_KEY = "f1a5564dfc20059ace98a12dafc792d8"
    }

}
